src/sentry/replays/query.py

```diff
@@ -153,7 +153,7 @@ def query_replays_dataset(
                 # Make sure we're not too old.
                 Condition(Column("finished_at"), Op.LT, end),
                 # Require non-archived replays.
-                Condition(Column("isArchived"), Op.EQ, 0),
+                Condition(Column("isArchived"), Op.IS_NULL),
```

Comment:
This will work given our current ingest patterns however this can break with behavior change.  For example if any row contains `is_archived=0` (which is a falsey value) `any(is_archived)` will return 0 which will break the `is null` conditional check.

We should use `max(is_archived)` and check the condition `(max(is_archived) = 0 or max(is_archived) is null)`.  Example use cases given below:

0 value case:

```sql
SELECT
    id,
    max(a) AS ma
FROM 
(
    SELECT
        1 AS id,
        0 AS a
)
GROUP BY id
HAVING isNull(ma) OR (ma < 1)

Query id: f69945a8-459d-4479-843d-6d3197400cb9

┌─id─┬─ma─┐
│  1 │  0 │
└────┴────┘
```

NULL value case:

```sql
SELECT
    id,
    max(a) AS ma
FROM 
(
    SELECT
        1 AS id,
        NULL AS a
)
GROUP BY id
HAVING isNull(ma) OR (ma < 1)

Query id: c2cd58aa-b758-4387-94bb-eb8ced5ab976

┌─id─┬─ma───┐
│  1 │ ᴺᵁᴸᴸ │
└────┴──────┘
```

1 value case:

```sql
SELECT
    id,
    max(a) AS ma
FROM 
(
    SELECT
        1 AS id,
        1 AS a
)
GROUP BY id
HAVING isNull(ma) OR (ma < 1)

Query id: 654c77a4-632a-4e6b-bfec-8b5f9acfaeda

Ok.

0 rows in set. Elapsed: 0.014 sec.
```

Reply:
ive just gone ahead and used the max function. lmk if that works.

Reply:
ah hold on logic isnt quite right.

Reply:
okay should be good!